railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -34,7 +34,7 @@ def build_stack
           # Must come after Rack::MethodOverride to properly log overridden methods
           middleware.use ::Rails::Rack::Logger, config.log_tags
           middleware.use ::ActionDispatch::ShowExceptions, show_exceptions_app
-          middleware.use ::ActionDispatch::DebugExceptions, app
+          middleware.use ::ActionDispatch::DebugExceptions, app, config.api_only
```

Comment:
@matthewd @spastorino I've refactored this to have a separate config option. You can see the related changes here: https://github.com/rails/rails/compare/ffa2a5de6414030a66ca2f61ca2ab725096b4158...jmbejar:rails-api-json-error-response

I initially tried having a boolean flag but it seems to fit better if we have a config setting with symbol values that describe the behavior of DebugExceptions. For now, we have `:default` and `:api` modes.

One drawback of this approach where we have separate config settings: if some app uses `api_only = false` and `debug_exception_response_format = :api` for some reason, it will end up having JSON responses for error requests while the app behaves as a default application.

This is a point of concern for me, but I understand that it would be caused by developer decisions since they will have more flexibility to define the behavior. In any case, I'm still pushing these commits to the PR to have more feedback and discussion.